src/chttpd_db.erl

```diff
@@ -225,9 +225,23 @@ maybe_flush_changes_feed(Acc0, Data, Len) ->
     },
     {ok, Acc}.
 
-handle_compact_req(Req, _) ->
-    Msg = <<"Compaction must be triggered on a per-shard basis in CouchDB">>,
-    couch_httpd:send_error(Req, 403, forbidden, Msg).
+handle_compact_req(#httpd{method='POST'}=Req, Db) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    case Req#httpd.path_parts of
```

Comment:
interesting, i took the original compact handler at https://github.com/apache/couchdb-couch/blob/aa5d63fce03f6bac18e7c1ffbbaf84f898baf010/src/couch_httpd_db.erl#L161 as an inspiration. will change it